code/parachain/runtime/common/src/xcmp.rs

```diff
@@ -132,16 +132,19 @@ impl<
 		Err(XcmError::TooExpensive)
 	}
 
+	// how does this refunding works? - add docs
 	fn refund_weight(&mut self, weight: Weight) -> Option<MultiAsset> {
 		if let Some(ref asset_location) = self.asset_location {
 			let fee = WeightToFeeConverter::weight_to_fee(&weight);
 			let fee = self.fee.min(fee);
 			let price = fee.saturating_mul(self.price) / self.fee;
+			if price <= 0 {
```

Comment:
price should never be 0. should recheck that and do 0.max(1)

Reply:
sure. this is correct. that is price_reduction, and fee is fee_reduction. better to rename for clarity. == 0